### PR TITLE
copy-tracking: add support for `diff --git`

### PR DESCRIPTION
**Pull request by fowles**
# Checklist

If applicable:
- [x] I have updated `CHANGELOG.md`
- [x] I have updated the documentation (README.md, docs/, demos/)
- [x] I have updated the config schema (cli/src/config-schema.json)
- [x] I have added tests to cover my changes

### CODE REVIEW

File: cli/src/diff_util.rs [hunk 1]
```diff
@@ -1098,23 +1104,40 @@ fn show_unified_diff_hunks(
...
-            let left_part = git_diff_part(&path, left_value)?;
-            let right_part = git_diff_part(&path, right_value)?;
+
+            let left_part = git_diff_part(&left_path, left_value)?;
+            let right_part = git_diff_part(&right_path, right_value)?;
+
+            // Skip the "delete" entry when there is a rename.
+            if right_part.mode.is_none() && copied_sources.contains(left_path.as_ref()) {
+                continue;
+            }
```
Comment (yuja) [inline cli/src/diff_util.rs:1134]: Slightly off topic, but I think this should be handled by `TreeDiffStream`/`Iterator`. From API standpoint, it's odd that source files are adjusted but the destinations aren't.

(Not meant for this PR)

File: cli/src/diff_util.rs [hunk 2]
```diff
@@ -1127,6 +1150,16 @@ pub fn show_git_diff(
                         writeln!(formatter, "index {left_hash}..{right_hash}")?;
                     }
                     (Some(left_mode), Some(right_mode)) => {
+                        if left_path != right_path {
+                            let operation = if to_tree.path_value(&left_path).unwrap().is_absent() {
```
Comment (yuja) [inline cli/src/diff_util.rs:1154]: nit: `path_value(..)?`

The return type of this closure will have to be `Result<(), DiffRenderError>`.

File: cli/tests/test_diff_command.rs
Comment (yuja) [inline cli/tests/test_diff_command.rs:1]: nit: Can you add "copy from"/"to" samples for `--git` and `--summary`?

**Review by yuja — APPROVED** [review]
Thanks!

File: cli/src/diff_util.rs (hunk 1 above)
Comment (fowles) [reply cli/src/diff_util.rs:1134]: I agree it is odd.  The core issue is that a copy is defined by

source -> still exists
source -> target

and a rename is defined by

source -> removed
source -> target

so the first case doesn't necessarily show up in a diff of the trees at all.

File: cli/src/diff_util.rs (hunk 2 above)
Comment (fowles) [reply cli/src/diff_util.rs:1154]: Done

File: cli/src/diff_util.rs (hunk 1 above)
Comment (yuja) [reply cli/src/diff_util.rs:1134]: I'm not sure if we'll need copy/rename distinction at store layer, but I think we can at least mask "source -> removed" entry by diff stream.

Maybe leave a TODO comment?

File: cli/tests/test_diff_command.rs
Comment (fowles) [reply cli/tests/test_diff_command.rs:1]: boom!  This found a misconfig in my git tracking stuff.  Nice one!!

File: cli/src/diff_util.rs (hunk 1 above)
Comment (fowles) [reply cli/src/diff_util.rs:1134]: I think I would prefer to wait for martin and google folks to have there meeting about how to redo the API here, cause the TODO would end up sprinkled in several places.

**Pull request merged**